iree/compiler/Dialect/Vulkan/Utils/TargetEnvUtils.cpp

```diff
@@ -212,6 +235,36 @@ const char *getTargetEnvForTriple(llvm::StringRef triple) {
     }>)";
   }
 
+  if (triple == "turing-t4-unknown-linux") {
+    return R"(#vk.target_env<
+     v1.2, r(133), [VK_KHR_storage_buffer_storage_class,
+       VK_NV_cooperative_matrix], NVIDIA:DiscreteGPU, {
+       cooperativeMatrixPropertiesNV = [{
+         mSize = 8: i32,
+         nSize = 8: i32,
+         kSize = 32: i32,
+         aType = i8,
+         bType = i8,
+         cType = i32,
+         resultType = i32,
+         scope = 3: i32
+       }, {
+         mSize = 8: i32,
+         nSize = 8: i32,
+         kSize = 16: i32,
+         aType = f16,
+         bType = f16,
+         cType = f16,
+         resultType = f16,
+         scope = 3: i32
+       }],
+       maxComputeSharedMemorySize = 49152: i32,
+       maxComputeWorkGroupInvocations = 1024: i32,
+       maxComputeWorkGroupSize = dense<[2147483647, 65535, 65535]> : vector<3xi32>,
+       subgroupFeatures = 63: i32,
+       subgroupSize = 32: i32
+    }>)";
+  }
```

Comment:
I think it should also support other fields described in https://github.com/google/iree/blob/main/iree/compiler/Dialect/Vulkan/IR/VulkanAttributes.td? it would be nice to have them registered (by reading the `vulkaninfo` dump) so the CodeGen naturally works for e.g., smaller bitwdiths, and we don't need to hunt this down again in the future. :)